### PR TITLE
fix #47249, pass `𝕃::AbstractLattice` to `fieldtype_nothrow`

### LINKED ISSUE #47249: `MethodError(f=Core.Compiler.:(⊑)` ... when testing some packages
Packages "ComoniconOptions" and "ManifoldsBase" errors when testing on master with:

```
Internal error: encountered unexpected error in runtime:
MethodError(f=Core.Compiler.:(⊑), args=(Core.Compiler.JLTypeLattice(), DataType, Core.Compiler.MaybeUndef(typ=Any)), world=0x000000000000148f)
jl_method_error_bare at /home/kc/julia/src/gf.c:1942
jl_method_error at /home/kc/julia/src/gf.c:1960
jl_lookup_generic_ at /home/kc/julia/src/gf.c:2613 [inlined]
ijl_apply_generic at /home/kc/julia/src/gf.c:2628
⊑ at ./compiler/typelattice.jl:331
⊑ at ./compiler/typelattice.jl:309 [inlined]
⊑ at ./compiler/typelattice.jl:249 [inlined]
⊑ at ./compiler/typelattice.jl:220 [inlined]
⊑ at ./compiler/abstractlattice.jl:168 [inlined]
fieldtype_nothrow at ./compiler/tfuncs.jl:1242
_builtin_nothrow at ./compiler/tfuncs.jl:1863
builtin_nothrow at ./compiler/tfuncs.jl:2103 [inlined]
builtin_effects at ./compiler/tfuncs.jl:2088
stmt_effect_flags at ./compiler/optimize.jl:261
check_effect_free! at ./compiler/ssair/inlining.jl:1219 [inlined]
process_simple! at ./compiler/ssair/inlining.jl:1270
assemble_inline_todo! at ./compiler/ssair/inlining.jl:1620
ssa_inlining_pass! at ./compiler/ssair/inlining.jl:78
jfptr_ssa_inlining_passNOT._15057 at /home/kc/julia/usr/lib/julia/sys.so (unknown line)
_jl_invoke at /home/kc/julia/src/gf.c:2431 [inlined]
ijl_apply_generic at /home/kc/julia/src/gf.c:2632
run_passes at ./compiler/optimize.jl:539
run_passes at ./compiler/optimize.jl:554 [inlined]
optimize at ./compiler/optimize.jl:503 [inlined]
_typeinf at ./compiler/typeinfer.jl:271
typeinf at ./compiler/typeinfer.jl:215
typeinf_edge at ./compiler/typeinfer.jl:952
abstract_call_method at ./compiler/abstractinterpretation.jl:609
```

PkgEval log example: https://s3.amazonaws.com/julialang-reports/nanosoldier/pkgeval/by_hash/c6440b9_vs_d0156b5/ComoniconOptions.primary.log

### PR DESCRIPTION
**Pull request by aviatesk**

### CODE REVIEW

**Comment by aviatesk** [comment]
@nanosoldier `runtests([ComoniconOptions", "ManifoldsBase"], vs = ":master")`

**Comment by aviatesk** [comment]
@nanosoldier `runtests([ComoniconOptions", "ManifoldsBase"], vs = ":master")`

File: base/compiler/abstractinterpretation.jl [hunk 1]
```diff
@@ -1837,19 +1837,18 @@ function abstract_call_opaque_closure(interp::AbstractInterpreter,
...
     sigT = (unwrap_unionall(tt)::DataType).parameters[1]
     match = MethodMatch(sig, Core.svec(), closure.source, sig <: rewrap_unionall(sigT, tt))
+    ⊑ₚ = ⊑(ipo_lattice(interp))
     const_result = nothing
     if !result.edgecycle
         const_call_result = abstract_call_method_with_const_args(interp, result,
             nothing, arginfo, si, match, sv)
         if const_call_result !== nothing
-            if const_call_result.rt ⊑ rt
+            if const_call_result.rt ⊑ₚ rt
```
Comment (aviatesk) [inline base/compiler/abstractinterpretation.jl:1846]: It turns out that the `ManifoldsBase` test suite needs this change also.

**Review by Keno — APPROVED** [review]

**Comment by DilumAluthge** [comment]
I'm seeing test failures with the message `ipo` not defined.

File: base/compiler/abstractinterpretation.jl (hunk 1 above)
Comment (aviatesk) [reply base/compiler/abstractinterpretation.jl:1850]: Ah, I mean, the corresponding change in `abstract_invoke`. These changes in `abstract_call_opaque_closure` are driven by that.

**Comment by aviatesk** [comment]
@nanosoldier `runtests([ComoniconOptions", "ManifoldsBase"], vs = ":master")`

**Pull request merged**